tests/integration/test_analysisd/test_signature_id/test_valid_signature_id.py

```diff
@@ -0,0 +1,141 @@
+t1_configurations = load_configuration_template(configurations_path, t1_configuration_parameters,
+                                                t1_configuration_metadata)
+
+
+# Tests
+@pytest.mark.tier(level=1)
+@pytest.mark.parametrize('configuration, metadata', zip(t1_configurations, t1_configuration_metadata), ids=t1_case_ids)
+def test_null_signature_id(configuration, metadata, set_wazuh_configuration, truncate_monitored_files,
+                           restart_wazuh_module):
+    '''
+    description: Check that when a rule has an valid signature ID value assigned to the if_sid option, the rule is
+                 not ignored.
+
+    test_phases:
+        - Copy custom rule file into manager
+        - Restart manager
+        - Check logs
+
+    wazuh_min_version: 4.4.0
+
+    tier: 1
+
+    parameters:
+        - configuration:
+            type: dict
+            brief: Configuration loaded from `configuration_template`.
+        - metadata:
+            type: dict
+            brief: Test case metadata.
+        - set_wazuh_configuration:
+            type: fixture
+            brief: Set wazuh configuration.
+        - truncate_monitored_files:
+            type: fixture
+            brief: Truncate all the log files and json alerts files before and after the test execution.
+
+    assertions:
+        - Check that wazuh starts
+        - Check ".*Signature ID '(\\d*)' was not found and will be ignored in the 'if_sid'.* of rule '(\\d*)'" event
+        - Check ".*wazuh-testrule.*Empty 'if_sid' value. Rule '(\\d*)' will be ignored.*"
+
+    input_description:
+        - The `configuration_signature_id_values.yaml` file provides the module configuration for
+          this test.
+        - The `cases_signature_id_values.yaml` file provides the test cases.
+    '''
+
+    rules_file_path = os.path.join(RULES_SAMPLE_PATH, metadata['rules_file'])
+    wazuh_log_monitor = FileMonitor(LOG_FILE_PATH)
+
+    # Copy rules to manager folder
+    copy(rules_file_path, CUSTOM_RULES_PATH)
+    chown(os.path.join(CUSTOM_RULES_PATH, metadata['rules_file']), WAZUH_UNIX_USER, WAZUH_UNIX_GROUP)
+
+    # Restart wazuh for changes to take effect
+    control_service(action='restart')
+
+    # Check logs
+    with pytest.raises(TimeoutError):
+        wazuh_log_monitor.start(timeout=T_5, callback=generate_monitoring_callback(CB_SID_NOT_FOUND),
+                                error_message=ERR_MSG_SID_NOT_FOUND)
+    with pytest.raises(TimeoutError):
+        wazuh_log_monitor.start(timeout=T_5, callback=generate_monitoring_callback(CB_EMPTY_IF_SID_RULE_IGNORED),
+                                error_message=ERR_MSG_EMPTY_IF_SID)
+    with pytest.raises(TimeoutError):
+        wazuh_log_monitor.start(timeout=T_5, callback=generate_monitoring_callback(CB_INVALID_IF_SID_RULE_IGNORED),
+                                error_message=ERR_MSG_INVALID_IF_SID)
```

Comment:
Why do you add 3 times `pytest.raises(TimeoutError)`? 
Is it not enough to put only one where inside you add the asserts?

Reply:
Fixed in https://github.com/wazuh/wazuh-qa/pull/3649/commits/106a8dc5732d09012d28419114e4e56478bfdded